sdk/webpubsub/Microsoft.Azure.WebJobs.Extensions.WebPubSub/src/Services/WebPubSubRequestExtensions.cs

```diff
@@ -151,12 +151,10 @@ internal static bool IsValidSignature(this WebPubSubConnectionContext connection
             if (!string.IsNullOrEmpty(connectionStates))
             {
                 var states = new Dictionary<string, object>();
-                var parsedStates = Encoding.UTF8.GetString(Convert.FromBase64String(connectionStates));
-                var statesObj = JsonDocument.Parse(parsedStates);
+                var statesObj = JsonDocument.Parse(Convert.FromBase64String(connectionStates));
                 foreach (var item in statesObj.RootElement.EnumerateObject())
                 {
-                    // Use ToString() to set pure value without ValueKind.
-                    states.Add(item.Name, item.Value.ToString());
+                    states.Add(item.Name, item.Value);
```

Comment:
Nit - I think you could accomplish the same thing with
```C#
var json = Convert.FromBase64String(connectionStates);
var states = JsonSerializer.Deserialize<Dictionary<string, JsonElement>>(json);
```
which might have more error handling around what happens in edge cases like the root element not being an object?  I'm not sure if there's a measurable perf difference between the two approaches though.